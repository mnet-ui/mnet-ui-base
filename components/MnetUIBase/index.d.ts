import * as React from "react";
import { ThemeType } from "../../themes";

export interface MnetUIBaseProps {
  cssVars?: boolean;
  full?: boolean;
  plain?: boolean;
  theme?: ThemeType;
  themeMode?: "dark" | "light";
  userAgent?: string;
}

declare const MnetUIBase: React.ComponentClass<MnetUIBaseProps & JSX.IntrinsicElements['div']>;

export { MnetUIBase };
