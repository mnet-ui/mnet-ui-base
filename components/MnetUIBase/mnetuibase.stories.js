"use strict";

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _mnetUiBase = require("mnet-ui-base");

var _grommetIcons = require("grommet-icons");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return _react["default"].createElement(_mnetUiBase.MnetUIBase, {
    theme: customTheme
  }, _react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, _react["default"].createElement(_mnetUiBase.Anchor, {
    icon: _react["default"].createElement(_grommetIcons.Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_mnetUiBase.MnetUIBase, {
    plain: true
  }, _react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, _react["default"].createElement("p", null, "Plain MnetUIBase"))), _react["default"].createElement(_mnetUiBase.MnetUIBase, null, _react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium"
  }, _react["default"].createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return _react["default"].createElement(_mnetUiBase.MnetUIBase, {
    theme: _mnetUiBase.mnet,
    cssVars: true
  }, _react["default"].createElement(_mnetUiBase.Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, _react["default"].createElement(_mnetUiBase.Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), _react["default"].createElement(_mnetUiBase.Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

(0, _react2.storiesOf)('MnetUIBase', module).add('Plain', function () {
  return _react["default"].createElement(Plain, null);
}).add('Theme', function () {
  return _react["default"].createElement(Themed, null);
}).add('Vars', function () {
  return _react["default"].createElement(MnetUIBaseVars, null);
});