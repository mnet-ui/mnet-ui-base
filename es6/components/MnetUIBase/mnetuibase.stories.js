import React from 'react';
import { storiesOf } from '@storybook/react';
import { mnet, MnetUIBase, Anchor, Box } from 'mnet-ui-base';
import { Add } from "grommet-icons/es6/icons/Add";
var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return React.createElement(MnetUIBase, {
    theme: customTheme
  }, React.createElement(Box, {
    pad: "medium"
  }, React.createElement(Anchor, {
    icon: React.createElement(Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return React.createElement(React.Fragment, null, React.createElement(MnetUIBase, {
    plain: true
  }, React.createElement(Box, {
    pad: "medium"
  }, React.createElement("p", null, "Plain MnetUIBase"))), React.createElement(MnetUIBase, null, React.createElement(Box, {
    pad: "medium"
  }, React.createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return React.createElement(MnetUIBase, {
    theme: mnet,
    cssVars: true
  }, React.createElement(Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, React.createElement(Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), React.createElement(Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

storiesOf('MnetUIBase', module).add('Plain', function () {
  return React.createElement(Plain, null);
}).add('Theme', function () {
  return React.createElement(Themed, null);
}).add('Vars', function () {
  return React.createElement(MnetUIBaseVars, null);
});