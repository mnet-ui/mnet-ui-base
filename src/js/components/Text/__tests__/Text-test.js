import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { axe } from 'jest-axe';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { MnetUIBase } from '../../MnetUIBase';
import { Text } from '..';

test('should have no accessibility violations', async () => {
  const { container } = render(
    <Grommet>
      <Text a11yTitle="test"> Example</Text>
    </Grommet>,
  );
  const results = await axe(container);
  expect(results).toHaveNoViolations();
  expect(container.firstChild).toMatchSnapshot();
});

test('renders', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text>text</Text>
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('accepts ref', () => {
  const ref = React.createRef();
  const component = renderer.create(
    <Grommet>
      <Text ref={ref}>text</Text>
    </Grommet>,
    { createNodeMock: el => el },
  );
  expect(ref.current).not.toBeNull();
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders size', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text size="xsmall" />
      <Text size="small" />
      <Text size="medium" />
      <Text size="large" />
      <Text size="xlarge" />
      <Text size="xxlarge" />
      <Text size="2xl" />
      <Text size="3xl" />
      <Text size="4xl" />
      <Text size="5xl" />
      <Text size="6xl" />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders textAlign', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text textAlign="start" />
      <Text textAlign="center" />
      <Text textAlign="end" />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders margin', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text margin="small" />
      <Text margin="medium" />
      <Text margin="large" />
      <Text margin="none" />
      <Text margin={{ vertical: 'small' }} />
      <Text margin={{ horizontal: 'small' }} />
      <Text margin={{ bottom: 'small' }} />
      <Text margin={{ top: 'small' }} />
      <Text margin={{ left: 'small' }} />
      <Text margin={{ right: 'small' }} />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';

test('renders truncate', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text truncate={false}>{LONG}</Text>
      <Text truncate>{LONG}</Text>
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders color', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text color="status-critical" />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders tag', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text as="div" />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('proxies tag', () => {
  const tagComponent = renderer.create(
    <MnetUIBase>
      <Text tag="div" />
    </MnetUIBase>,
  );
  const asComponent = renderer.create(
    <MnetUIBase>
      <Text as="div" />
    </MnetUIBase>,
  );
  expect(tagComponent.toJSON()).toEqual(asComponent.toJSON());
});

test('renders weight', () => {
  const component = renderer.create(
    <MnetUIBase>
      <Text weight="normal" />
      <Text weight="bold" />
    </MnetUIBase>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
