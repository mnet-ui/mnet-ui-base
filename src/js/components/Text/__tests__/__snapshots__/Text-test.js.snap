// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts ref 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    text
  </span>
</div>
`;

exports[`renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    text
  </span>
</div>
`;

exports[`renders color 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  color: #e35e59;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
</div>
`;

exports[`renders margin 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c2 {
  margin: 16px;
  font-size: 12px;
  line-height: 16px;
}

.c3 {
  margin: 32px;
  font-size: 12px;
  line-height: 16px;
}

.c4 {
  margin: 0px;
  font-size: 12px;
  line-height: 16px;
}

.c5 {
  margin-top: 8px;
  margin-bottom: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c6 {
  margin-left: 8px;
  margin-right: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c7 {
  margin-bottom: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c8 {
  margin-top: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c9 {
  margin-left: 8px;
  font-size: 12px;
  line-height: 16px;
}

.c10 {
  margin-right: 8px;
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
  <span
    className="c5"
  />
  <span
    className="c6"
  />
  <span
    className="c7"
  />
  <span
    className="c8"
  />
  <span
    className="c9"
  />
  <span
    className="c10"
  />
</div>
`;

exports[`renders size 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  font-size: 12px;
  line-height: 16px;
}

.c1 {
  font-size: 8px;
  line-height: 12px;
}

.c2 {
  font-size: 9.333333333333334px;
  line-height: 13.333333333333334px;
}

.c4 {
  font-size: 14.666666666666666px;
  line-height: 18.666666666666668px;
}

.c5 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
}

.c6 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
}

.c7 {
  font-size: 42px;
  line-height: 48px;
}

.c8 {
  font-size: 54px;
  line-height: 60px;
}

.c9 {
  font-size: 70px;
  line-height: 76px;
}

.c10 {
  font-size: 90px;
  line-height: 96px;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
  <span
    className="c4"
  />
  <span
    className="c5"
  />
  <span
    className="c6"
  />
  <span
    className="c6"
  />
  <span
    className="c7"
  />
  <span
    className="c8"
  />
  <span
    className="c9"
  />
  <span
    className="c10"
  />
</div>
`;

exports[`renders tag 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`renders textAlign 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  text-align: left;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  text-align: center;
}

.c3 {
  font-size: 12px;
  line-height: 16px;
  text-align: right;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
  <span
    className="c3"
  />
</div>
`;

exports[`renders truncate 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
  <span
    className="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </span>
</div>
`;

exports[`renders weight 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  font-weight: normal;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  font-weight: bold;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c2"
  />
</div>
`;

exports[`should have no accessibility violations 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
>
  <span
    aria-label="test"
    class="c1"
  >
     Example
  </span>
</div>
`;
